app/lib/smaxtec_api.rb

```diff
@@ -31,6 +56,11 @@ def update_sensor_readings
 
 
   def smaxtec_sensor_readings(sensor)
```

Comment:
Yeah, I wasn't sure about this. I didn't want too much repition, so I mixed the two use cases, but I guess you are right: they should be separated into own methods.